clients/src/test/java/org/apache/kafka/clients/producer/MockProducerTest.java

```diff
@@ -152,15 +154,17 @@ public void shouldBeginTransactions() {
     @Test(expected = IllegalStateException.class)
     public void shouldThrowOnSendOffsetsToTransactionIfTransactionsNotInitialized() {
         buildMockProducer(true);
-        producer.sendOffsetsToTransaction(null, null);
+        String groupId = null;
```

Comment:
To ensure that we are really testing the state machine as expected, we should provide a valid groupId. Similarly below.

Reply:
I guess the `MockProducer` is putting the null check later than the state check. I could re-order them if you feel better about it.

Reply:
The purpose of this test is to raise an exception if the transaction hasn't been initialized. But we are also providing an invalid groupId. To make sure we hit the right error case, we should provide a valid groupId.